rasa/core/events/__init__.py

```diff
@@ -81,8 +81,20 @@ class Event:
 
     type_name = "event"
 
-    def __init__(self, timestamp: Optional[float] = None):
-        self.timestamp = timestamp if timestamp else time.time()
+    def __init__(
+        self, timestamp: Optional[float] = None, metadata: Optional[Dict] = None
+    ):
+        self.timestamp = timestamp or time.time()
+        self._metadata = metadata or {}
+
+    @property
+    def metadata(self):
+        # needed for backwards compatibility <1.0.0 - previously pickled events
```

Comment:
this isn't the right version, this should be the most recent one no?

Reply:
I actually moved this method from `BotUttered`, so the comment isn't mine. I saw that the changelog mentions that we're not doing pickle serialization anymore since 1.4.0, so that means that the comment should specify <1.4.0, right?

Reply:
I think it depends on the phrasing 😅 - we don't do any pickling anymore, but not sure if there are other tracker store implementations out there that still use pickling. as long it is clear what we mean I am good

Reply:
Updated the comment to:
```
        # Needed for compatibility for Rasa versions <1.4.0.
        # Previous versions of Rasa serialized trackers using the pickle
        # module.  For the moment, Rasa still supports loading these serialized
        # trackers with pickle, but will use JSON in any subsequent save
        # operations.  Versions of trackers serialized with pickle won't
        # include the `_metadata` attribute in their events, so it is necessary
        # to define this getter in case the attribute does not exist.
        # For more information see CHANGELOG.rst.
```